projects/lexical_shared/lib/lexical/document/path.ex

```diff
@@ -125,6 +127,35 @@ defmodule Lexical.Document.Path do
     end
   end
 
+  @spec language_id_from_path(Lexical.path()) :: String.t()
+  def language_id_from_path(path) do
```

Comment:
Language id seems to be a property of `Lexical.Document` and not really related to paths. This module is really about path manipulation and conversion to and from URIs. 
I also think you can move this to `Lexical.Document` and use the URI module to get the path.

Reply:
Agreed, moved this function from `Document.Path` to `Document`.

I'd prefer this function assume its input is a path, and have callers invoke `Document.Path.ensure_path/1` should they need to disambiguate it.